tools/pre-commit

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
```

Comment:
`/home/runner/work/nuttx/nuttx/nuttx/tools/pre-commit: error: execute permissions detected!`

Reply:
because we can't add .sh suffix to pre-commit since git require the file name must be pre-commit